rust/lance-linalg/src/distance/cosine.rs

```diff
@@ -211,7 +282,7 @@ where
     let y_sq = dot(y, y);
     let xy = dot(x, y);
     // 1 - xy / (sqrt(x_sq) * sqrt(y_sq))
-    1.0 - xy / x_norm * y_sq.sqrt()
+    1.0 - xy / x_norm / y_sq.sqrt()
```

Comment:
This seems wrong to me.

Reply:
Have you added tests that failed previously? It sounds like we need much better testing of these kernels if it was incorrect before.

Reply:
Why should this be division?

Cosine similarity is `A dot B / (||A|| * ||B||)`?

Reply:
nvm, I see the problem now.  Either it needs to be `1.0 - xy / (x_norm * y_sq.sqrt())` or it needs to be what you have.

Reply:
I think this is non-SIMD fallback for implementation f32?